algorithms/src/prf/blake2xs.rs

```diff
@@ -0,0 +1,344 @@
+        // • “Key length” is set to 0 (even if the root hash was keyed)
+        const KEY_LENGTH: u8 = 0u8;
+        // • “Fanout” is set to 0 (unlimited)
+        const FANOUT: u8 = 0u8;
+        // • “Maximal depth” is set to 0
+        const DEPTH: u8 = 0u8;
+        // • “Leaf maximal byte length” is set to 32 for BLAKE2Xs, and 64 for BLAKE2Xb
+        const LEAF_LENGTH: u32 = 32u32;
+        // • “Node depth” is set to 0 (leaves)
+        const NODE_DEPTH: u8 = 0u8;
+        // • “Inner hash byte length” is set to 32 for BLAKE2Xs and 64 for BLAKE2Xb
+        const INNER_LENGTH: u8 = 32u8;
+        // • Other fields are left to the same values as in the underlying BLAKE2 instance
+        const SALT: u64 = 0u64;
+
+        Self::parameter_block::<
+            DIGEST_LENGTH,
+            KEY_LENGTH,
+            FANOUT,
+            DEPTH,
+            LEAF_LENGTH,
+            NODE_OFFSET,
+            XOF_DIGEST_LENGTH,
+            NODE_DEPTH,
+            INNER_LENGTH,
+            SALT,
+            PERSONALIZATION,
+        >()
+    }
+
+    /// Returns the parameter block for BLAKE2s.
+    #[rustfmt::skip]
+    pub const fn blake2s_parameter_block() -> [u32; 8] {
+        Self::parameter_block::<32u8, 0u8, 1u8, 1u8, 0u32, 0u32, 0u16, 0u8, 0u8, 0u64, 0u64>()
+    }
+
+    #[rustfmt::skip]
+    pub const fn parameter_block<
+        const DIGEST_LENGTH: u8,
+        const KEY_LENGTH: u8,
+        const FANOUT: u8,
+        const DEPTH: u8,
+        const LEAF_LENGTH: u32,
+        const NODE_OFFSET: u32,
+        const XOF_DIGEST_LENGTH: u16,
+        const NODE_DEPTH: u8,
+        const INNER_LENGTH: u8,
+        const SALT: u64,
+        const PERSONALIZATION: u64,
+    >() -> [u32; 8] {
+        [
+            // Offset 0 - Digest length || Key length || Fanout || Depth
+            u32::from_le_bytes([DIGEST_LENGTH, KEY_LENGTH, FANOUT, DEPTH]),
+            // Offset 4 - Leaf length
+            LEAF_LENGTH,
+            // Offset 8 - Node offset
+            NODE_OFFSET,
+            // Offset 12 - XOF digest length || Node depth || Inner length
+            u32::from_le_bytes([XOF_DIGEST_LENGTH as u8, (XOF_DIGEST_LENGTH >> 8) as u8, NODE_DEPTH, INNER_LENGTH]),
+            // Offset 16 - Salt
+            u32::from_le_bytes([SALT as u8, (SALT >> 8) as u8, (SALT >> 16) as u8, (SALT >> 24) as u8]),
+            // Offset 20 - Salt (continued)
+            u32::from_le_bytes([(SALT >> 32) as u8, (SALT >> 40) as u8, (SALT >> 48) as u8, (SALT >> 56) as u8]),
+            // Offset 24 - Personalization
+            u32::from_le_bytes([PERSONALIZATION as u8, (PERSONALIZATION >> 8) as u8, (PERSONALIZATION >> 16) as u8, (PERSONALIZATION >> 24) as u8]),
+            // Offset 24 - Personalization (continued)
+            u32::from_le_bytes([(PERSONALIZATION >> 32) as u8, (PERSONALIZATION >> 40) as u8, (PERSONALIZATION >> 48) as u8, (PERSONALIZATION >> 56) as u8]),
+        ]
+    }
+}
+
+#[cfg(test)]
+mod tests {
```

Comment:
Reference test vectors available here: https://raw.githubusercontent.com/BLAKE2/BLAKE2/master/testvectors/blake2-kat.json

Reply:
Thanks! Added the test vectors to the unit tests